enhancements/microshift/microshift-apiserver-certs.md

```diff
@@ -0,0 +1,454 @@
+---
+title: microshift-apiserver-certs
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@fzdarsky, MicroShift architect"
+  - "@ggiguash, MicroShift contributor"
+  - "@stlaz, Security specialist"
+  - "@pmtk, MicroShift contributor"
+  - "@copejon, MicroShift contributor"
+  - "@deads2k, OpenShift architect"
+  - "@benluddy, OpenShift API server team"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - N/A
+creation-date: 2023-01-20
+last-updated: 2023-01-20
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-716
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# MicroShift API server certificates
+## Summary
+This enhancement proposes a strategy to manage kube API server certificates
+in all environments where MicroShift might be deployed, both externally
+(accessible through kubeconfig) and internally (pods reaching the API server).
+
+> Anytime the document mentions API server, it refers to kube API server.
+
+MicroShift has a wide variety of different use cases when it comes to API
+server connectivity, we can find environments where the cluster needs to be
+reachable with the IP, with the hostname, by some other name having DNS, or
+a combination of them.
+
+## Motivation
+As of today, MicroShift does not cover all authentication requirements. It
+needs to be able to authenticate API server connections using the node IP
+address, the hostname, or any other domain name with DNS resolution. It
+shares certificate handling with OCP, which is a bit different in terms of
+deployment.
+
+Unlike OCP, MicroShift does not have a load balancer to access API server, and
```

Comment:
Was an alternative that introduces a load balancer considered?

Reply:
We've been trying to avoid that. With just 1 host, there's nothing to "balance."